skrub/tests/test_string_encoder.py

```diff
@@ -0,0 +1,142 @@
+import pytest
+from sklearn.decomposition import TruncatedSVD
+from sklearn.feature_extraction.text import HashingVectorizer, TfidfVectorizer
+from sklearn.pipeline import Pipeline
+
+from skrub import _dataframe as sbd
+from skrub._string_encoder import StringEncoder
+
+
+@pytest.fixture
+def encode_column(df_module):
+    corpus = [
+        "this is the first document",
+        "this document is the second document",
+        "and this is the third one",
+        "is this the first document",
+    ]
+
+    return df_module.make_column("col1", corpus)
+
+
+def test_encoding(encode_column, df_module):
+    ngram_range = (3, 4)
+    analyzer = "char_wb"
+    n_components = 2
+
+    #### tfidf vectorizer
+    pipe = Pipeline(
+        [
+            ("tfidf", TfidfVectorizer(ngram_range=ngram_range, analyzer=analyzer)),
+            ("tsvd", TruncatedSVD(n_components=n_components)),
+        ]
+    )
+    check = pipe.fit_transform(sbd.to_numpy(encode_column))
+
+    names = [f"col1_{idx}" for idx in range(2)]
+
+    check_df = df_module.make_dataframe(dict(zip(names, check.T)))
+
+    se = StringEncoder(
+        n_components=n_components,
+        vectorizer="tfidf",
+        ngram_range=ngram_range,
+        analyzer=analyzer,
+    )
+    result = se.fit_transform(encode_column)
+
+    # Converting dtypes to avoid nullable shenanigans
+    check_df = sbd.pandas_convert_dtypes(check_df)
+    result = sbd.pandas_convert_dtypes(result)
+
+    df_module.assert_frame_equal(check_df, result)
+
+
+def test_hashing(encode_column, df_module):
```

Comment:
I think this test is failing because the hashing vectorizer is not deterministic, if that's the case, I'm not sure how to handle it in the test other than exposing RNG everywhere. 

I didn't bother because we haven't decided whether to keep the hashing vectorizer in the first place.

Reply:
otherwise in the test you can just check the type and shape of the output and the content of `self.pipe`